app/controllers/shipyard/ApprenticeStorekeeperController.scala

```diff
@@ -0,0 +1,85 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.shipyard
+
+import config.ClaimAmounts
+import controllers.actions._
+import forms.shipyard.ApprenticeStorekeeperFormProvider
+import javax.inject.{Inject, Named}
+import models.Mode
+import navigation.Navigator
+import pages.ClaimAmount
+import pages.manufacturing.AluminiumApprenticePage
+import pages.shipyard.ApprenticeStorekeeperPage
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
+import uk.gov.hmrc.play.bootstrap.controller.FrontendBaseController
+import views.html.shipyard.ApprenticeStoreKeeperView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class ApprenticeStorekeeperController @Inject()(
+                                         override val messagesApi: MessagesApi,
+                                         sessionRepository: SessionRepository,
+                                         @Named("Generic") navigator: Navigator,
+                                         identify: UnauthenticatedIdentifierAction,
+                                         getData: DataRetrievalAction,
+                                         requireData: DataRequiredAction,
+                                         formProvider: ApprenticeStorekeeperFormProvider,
+                                         val controllerComponents: MessagesControllerComponents,
+                                         view: ApprenticeStoreKeeperView
+                                 )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+
+  val form: Form[Boolean] = formProvider()
+
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) {
+    implicit request =>
+
+      val preparedForm = request.userAnswers.get(ApprenticeStorekeeperPage) match {
+        case None => form
+        case Some(value) => form.fill(value)
+      }
+
+      Ok(view(preparedForm, mode))
+  }
+
+  def onSubmit(mode: Mode) = (identify andThen getData andThen requireData).async {
+    implicit request =>
+
+      form.bindFromRequest().fold(
+        (formWithErrors: Form[_]) =>
+          Future.successful(BadRequest(view(formWithErrors, mode))),
+
+        value => {
+          for {
+            updatedAnswers <- if (value) {
+              Future.fromTry(request.userAnswers.set(AluminiumApprenticePage, value)
```

Comment:
userAnswers.set(AluminiumApprenticePage) should this not be ApprenticeStoreKeeperPage?